hibernate/src/main/java/implementations/entitiesImpl/UserRoleDaoImpl.java

```diff
@@ -0,0 +1,26 @@
+package implementations.entitiesImpl;
+
+import com.becomejavasenior.UserRole;
+import com.becomejavasenior.dao.hibernatedao.UserRoleDao;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+
+import java.util.List;
+
+public class UserRoleDaoImpl extends AbstractDaoImpl implements UserRoleDao{
+    private static final Logger LOGGER = Logger.getLogger(UserRoleDaoImpl.class);
+
+    @Override
+    public UserRole getByPK(Integer id) {
+        Session session = getSession();
+        UserRole userRole = (UserRole) session.load(UserRole.class, id);
+        LOGGER.debug(UserRoleDaoImpl.class);
+        commitTransaction(session);
```

Comment:
in general you shouldn't control transaction by yourself, this work should be done by hibernate or transaction manager in case of JTA, every time you make a commit the first level cache is flushing and that may cause performance degradation